marlowe-playground-client/src/Simulation.purs

```diff
@@ -253,6 +278,79 @@ handleAction settings AnalyseReachabilityContract = do
 
 handleAction _ Save = pure unit
 
+setOraclePrice ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ -> HalogenM State Action ChildSlots Void m Unit
+setOraclePrice settings = do
+  exState <- use (_currentMarloweState <<< _executionState)
+  case exState of
+    Just es -> do
+      let
+        (Parties actions) = es.possibleActions
+      case Map.lookup (Role "kraken") actions of
+        Just acts -> do
+          case Array.head (Map.toUnfoldable acts) of
+            Just (Tuple (ChoiceInputId choiceId@(ChoiceId pair _) bounds) _) -> do
+              price <- getPrice settings "kraken" pair
+              liftEffect $ log ("setting choice " <> show choiceId <> " with price " <> show price)
+              handleAction settings (SetChoice choiceId price)
+            _ -> pure unit
+        Nothing -> pure unit
+    Nothing -> pure unit
+
+getPrice :: forall m. MonadAff m => SPSettings_ SPParams_ -> String -> String -> HalogenM State Action ChildSlots Void m BigInteger
+getPrice settings exchange pair = do
+  result <- runAjax (getPriceAjax settings exchange pair)
+  a <-
+    liftEffect case result of
+      NotAsked -> do
+        log "NotAsked"
+        pure 0.0
+      Loading -> do
+        log "Loading"
+        pure 0.0
+      Failure e -> do
+        log $ "Failure" <> errorToString e
+        pure 0.0
+      Success a -> pure a.price
+  let
+    price = fromMaybe (fromInt 0) (fromString (show (floor a)))
+  pure price
+
+getPriceAjax ::
+  forall m.
+  MonadError AjaxError m =>
+  MonadAff m =>
+  (SPSettings_ SPParams_) -> String -> String -> m { price :: Number }
+getPriceAjax settings exchange pair = do
```

Comment:
this function should be generated from the Haskell definition of the API, see https://github.com/input-output-hk/plutus/blob/master/marlowe-playground-server/app/PSGenerator.hs#L256 whatever is in that API will be in generated client code in `marlowe-playground-client/generated/Marlowe.purs`.

Reply:
in fact looking at your change to `API.API` I think you'll find that it is in the generated code (of course you need to run `$(nix-build -A marlowe-playground.server-invoker)/bin/marlowe-playground psgenerator ./marlowe-playground-client/generated` to refresh it